comicapi/genericmetadata.py

```diff
@@ -204,96 +252,186 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
         new_md.__post_init__()
         return new_md
 
-    def overlay(self, new_md: GenericMetadata) -> None:
-        """Overlay a metadata object on this one
-
-        That is, when the new object has non-None values, over-write them
-        to this one.
-        """
-
-        def assign(cur: str, new: Any) -> None:
-            if new is not None:
-                if isinstance(new, str) and len(new) == 0:
-                    if isinstance(getattr(self, cur), (list, set)):
-                        getattr(self, cur).clear()
-                    else:
-                        setattr(self, cur, None)
-                elif isinstance(new, (list, set)) and len(new) == 0:
-                    pass
-                else:
-                    setattr(self, cur, new)
+    def weblinks_dedupe(self, new: list[Url], cur: list[Url]) -> list[Url]:
+        if len(new) == 0:
+            return cur
+        if len(cur) == 0:
+            return new
+
+        new_set: set[str] = {n.url for n in new}
+        cur_set: set[str] = {c.url for c in cur}
+        new_set.update(cur_set)
+
+        return [parse_url(url) for url in new_set]
```

Comment:
This is mostly unnecessary. The Url object is a tuple and the url property (actually a method) only does basic null checks and concatenation no sanitizing is done therefore this can be deduplicated with a simple set.
```python
list(set(new).union(cur))
```